functions/Resolve-DbaNetworkName.ps1

```diff
@@ -114,13 +114,24 @@ Returns a custom object displaying InputName, ComputerName, IPAddress, DNSHostNa
 			    {
 				    Write-Verbose "Getting computer information from server $Computer via WMI (DCOM)"
 				    $conn = Get-WmiObject -ComputerName $Computer -Query "Select Name, Caption, DNSHostName, Domain FROM Win32_computersystem" -ErrorAction SilentlyContinue -Credential $Credential
-					if ( !$conn )
-					{
-						Write-Warning "No DCOM connection for WMI to $Computer"
+				}
+				
+				if (!$conn)
+				{
+					Write-Verbose "Cim and Wmi both failed. Defaulting to .NET"
+					$fqdn = ([System.Net.Dns]::GetHostEntry($Computer)).HostName
+					$hostname = $fqdn.Split(".")[0]
+					
+					$conn = [PSCustomObject]@{
+						Name = $Computer
```

Comment:
nice addition

but this object needs to have the same properties as the other one.